txnsync/bitmask.go

```diff
@@ -21,26 +21,26 @@ import (
 	"errors"
 )
 
+var errIndexOutOfBounds = errors.New("invalid bitmask: index out of bounds")
+var errIndexNotFound = errors.New("invalid bitmask: index not found")
+var errInvalidBitmaskType = errors.New("invalid bitmask type")
+
 //msgp:allocbound bitmask maxBitmaskSize
 type bitmask []byte
 
 // assumed to be in mode 0, sets bit at index to 1
-func (b *bitmask) SetBit(index int) {
+func (b *bitmask) setBit(index int) {
 	byteIndex := index/8 + 1
 	(*b)[byteIndex] |= 1 << (index % 8)
 }
 
-func (b *bitmask) EntryExists(index int, entries int) bool {
-	if len(*b) == 0 {
-		return false
-	}
-	if (*b)[0] != 0 {
-		b.expandBitmask(entries)
-	}
-	byteIndex := index/8 + 1
-	return byteIndex < len(*b) && ((*b)[byteIndex]&(1<<(index%8)) != 0)
-}
-
+// trimBitmask compresses the bitmask into one of the 4 types:
+// type 0: intput bitmask bit pos x b -> output bitmask bit pos x b
+// type 1: intput bitmask bit pos x b -> output bitmask bit pos x !b
+// type 2: stores the positions of bits where b = 1
```

Comment:
The method does not check for the original bitmask type. One less check for performance. 
However, the code only uses this on type 0 bitmask.

Reply:
I was referring to the comment you wrote on the following two lines. I'm not sure if they are helpful/accurate, can we remove them?

Reply:
Helpful: I believe it is helpful. Any comment which is accurate is helpful. 
Accurate: what do you think is not accurate? Which argument?

Reply:
I don't understand what `first b=1 pos A` means. I assume it means the first byte is for position A, which is inaccurate. If I'm wrong please update the comment to make the meaning more clear.

Next, saying `output bitmask byte 2,A/256,A%256,(B-A)/256,(B-A)%256` is not helpful since that simply rewrites the code in a comment and doesn't explain anything. A better comment would say something regarding using adjacent differences of indices where the bit is set to 1.

Reply:
I hope this is now clearer. Updated.